### PR TITLE
chenriks/cookies

### PR DESCRIPTION
**Pull request by chenriksson**
-Removes the BOM signature from clientaccesspolicy.xml
-Adds Secure attribute to ForceSSL cookie
-Adds HttpOnly attribute to TempData cookie when cleared

### CODE REVIEW

File: src/NuGetGallery/Web.config [hunk 1]
```diff
@@ -157,7 +157,6 @@
   <!-- Ensure only Admins may access elmah -->
   <location path="Admin" inheritInChildApplications="false">
     <system.web>
-      <httpCookies httpOnlyCookies="true" />
```
Comment (dtivel) [inline src/NuGetGallery/Web.config]: You're removing this because it's inherited from the section below?

Comment (chenriksson) [inline src/NuGetGallery/Web.config]: Yes, moved it to the global system.web section. This change results in 'HttpOnly' set on the TempData cookie when it gets cleared; it's already adding 'HttpOnly' when TempData is not empty.

**Comment by skofman1** [comment]
@blowdart ?

**Comment by blowdart** [comment]
LGTM

File: src/NuGetGallery/Authentication/ForceSslWhenAuthenticatedMiddleware.cs [hunk 2]
```diff
@@ -47,7 +47,12 @@ public ForceSslWhenAuthenticatedMiddleware(OwinMiddleware next, IAppBuilder app,
                 // Invoke the rest of the pipeline
                 await Next.Invoke(context);
 
-                var cookieOptions = new CookieOptions() { HttpOnly = true };
+                var cookieOptions = new CookieOptions()
+                {
+                    HttpOnly = true,
+                    Secure = context.Request.IsSecure
```
Comment (skofman1) [inline src/NuGetGallery/Authentication/ForceSslWhenAuthenticatedMiddleware.cs]: >Secure [](start = 20, length = 6)

what does this do?

**Comment by skofman1** [comment]
> <?xml version="1.0" encoding="utf-8"?>

anything changed in this file?

---
Refers to: src/NuGetGallery/Public/clientaccesspolicy.xml:1 in 059db77. [](commit_id = 059db77a378e67f1be1c0e73ccb7b11068b27dae, deletion_comment = True)

**Comment by skofman1** [comment]
:shipit:

File: src/NuGetGallery/Authentication/ForceSslWhenAuthenticatedMiddleware.cs (hunk 2 above)
Comment (chenriksson) [inline src/NuGetGallery/Authentication/ForceSslWhenAuthenticatedMiddleware.cs]: Adds the 'Secure' attribute to the ForceSsl cookie when going over HTTPS

**Comment by chenriksson** [comment]
@skofman1 re:clientaccesspolicy.xml: change is removing the UTF-8 signature

**Pull request merged**